### PR TITLE
fix:add array declarations to installer 

### LINKED ISSUE #1603: Fix:IcmsInstallWizard $pagenames property is deprecated in PHP 8.4
Creation of dynamic property IcmsInstallWizard::$pagesNames is deprecated in C:\Users\david\sites\unify\install\class\IcmsInstallWizard.php on line 62

### PR DESCRIPTION
**Pull request by fiammybe**
This resolves the deprecated dynamic properties error during installation

fixes #1603

### CODE REVIEW

**Review by skenow — COMMENTED** [review]
Do we want to go ahead and change all properties to be declared without using 'var'?

**Comment by fiammybe** [comment]
I'll update the declarations to be more in line with newer PHP standards. That should placate codeclimate :-)

File: htdocs/install/class/IcmsInstallWizard.php [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
+	protected array $pagesTitles = array();
+	protected int $currentPage = 0;
+	protected string $currentPageName;
+	protected int $lastpage;
+	protected int $secondlastpage;
+	protected string $language = 'english';
+	protected bool $no_php5 = false;
+	protected bool $safe_mode = false;
+
+	function xoInit(): bool {
```
[the remark this answers is not in the document]
Comment (skenow) [reply htdocs/install/class/IcmsInstallWizard.php]: Hmm - is there a class/method we already have for accessing super globals? https://phpmd.org/rules/controversial.html

File: htdocs/install/class/IcmsInstallWizard.php [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
-	var $titles = array();
-	var $currentPage = 0;
-	var $lastpage;
-	var $secondlastpage;
-	var $language = 'english';
-	var $no_php5 = false;
-	var $safe_mode = false;
-
-	function xoInit() {
+	protected array $pages = array();
```
[the remark this answers is not in the document]
Comment (skenow) [reply htdocs/install/class/IcmsInstallWizard.php]: Trying to figure out which 'array' codeclimate is complaining about .

With the type declaration, do we also need the initialization?

protected array $pages;

File: htdocs/install/class/IcmsInstallWizard.php [hunk 3]
```diff
@@ -1,24 +1,27 @@
...
+	public array $titles = array();
+	public int $currentPage = 0;
+	public string $currentPageName;
+	public string $lastpage;
+	public string $secondlastpage = '';
+	public string $language = 'english';
+	public bool $no_php5 = false;
+	public bool $safe_mode = false;
+
+	public function xoInit(): bool {
```
[the remark this answers is not in the document]
Comment (fiammybe) [reply htdocs/install/class/IcmsInstallWizard.php:16]: Nothing we can do about that for the moment

**Pull request merged**